src/gui/editors/SongEditor.cpp

```diff
@@ -542,65 +529,33 @@ void SongEditor::keyPressEvent( QKeyEvent * ke )
 	}
 	else if (ke->key() == Qt::Key_0 && ke->modifiers() & Qt::ControlModifier)
 	{
-		m_zoomingLogValue = ZOOM_INI;
-		m_zoomingLinearModel->setValue(scaleToLinearZoom(m_zoomingLogValue));
+		m_zoomingModel->reset();
 	}
 	else
 	{
 		QWidget::keyPressEvent(ke);
 	}
 }
 
-void SongEditor::keyReleaseEvent(QKeyEvent* ke)
-{
-	if (ke->modifiers() & Qt::ControlModifier || ke->modifiers() & Qt::ShiftModifier) { m_iniBar = -1; }
-	QWidget::keyReleaseEvent(ke);
-}
+
+
 
 void SongEditor::wheelEvent( QWheelEvent * we )
 {
 	if ((we->modifiers() & Qt::ControlModifier) && (position(we).x() > m_trackHeadWidth))
 	{
-		float zVar = 0.1f;
-		if (we->modifiers() & Qt::ShiftModifier) { zVar = 0.01f; }
-
-		int z = m_zoomingLogValue;
-		if (we->angleDelta().y() > 0)
-		{
-			z = z * (1.0f + zVar);
-			int t = static_cast<int>(
-				std::ceil((std::floor(z / 100.0f * DEFAULT_PIXELS_PER_BAR) + 1) * 100.0f / DEFAULT_PIXELS_PER_BAR));
-			z = std::max(z, t);
-		}
-		else
-		{
-			z = z * (1.0f - zVar);
-			int t = std::floor((std::ceil(z / 100.0f * DEFAULT_PIXELS_PER_BAR) - 1) * 100.0f / DEFAULT_PIXELS_PER_BAR);
-			z = std::min(z, t);
-		}
-
-		z = std::clamp(z, ZOOM_MIN, ZOOM_MAX);
-
 		int x = position(we).x() - m_trackHeadWidth;
-		// save the pixels up to the cursor next bar when starting zooming
-		if (m_iniBar == -1)
-		{
-			m_iniBar = static_cast<int>(x / pixelsPerBar());
-		}
+		// bar based on the mouse x-position where the scroll wheel was used
+		int bar = x / pixelsPerBar();
 
-		// update slider & model with new zooming value (pixelsPerBar is recalcutaled based on new zoom value)
-		m_zoomingLogValue = z;
-		m_zoomingLinearModel->setValue(scaleToLinearZoom(z));
+		// move zoom slider (pixelsPerBar will change automatically)
+		int step = we->modifiers() & Qt::ShiftModifier ? 1 : 5;
+		int direction = we->angleDelta().y() > 0 ? 1 : -1;
+		m_zoomingModel->incValue(step * direction);
 
 		// scroll to zooming around cursor's bar
 		int newBar = static_cast<int>(x / pixelsPerBar());
-
-		// scroll, if necessary
-		if (m_iniBar != newBar)
-		{
-			m_leftRightScroll->setValue(m_leftRightScroll->value() + m_iniBar - newBar);
-			m_iniBar = newBar;
```

Comment:
Since you write to `m_iniBar` every time it moves to another bar, it serves no purpose any longer.